contrib/jmx-metrics/src/main/groovy/io/opentelemetry/contrib/jmxmetrics/MBeanHelper.groovy

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.contrib.jmxmetrics
+
+import javax.management.MBeanServerConnection
+import javax.management.ObjectName
+import java.util.logging.Logger
+
+/**
+ * A helper for easy {@link GroovyMBean} querying, creation, and protected attribute access.
+ *
+ * Basic query functionality is provided by the static queryJmx methods, while MBeanHelper
+ * instances operate on an underlying GroovyMBean list field.
+ *
+ * Specifying if a single underlying GroovyMBean is expected or of interest is done via the
+ * isSingle argument, where resulting attribute values are only returned for the first
+ * match.
+ *
+ * Intended to be used via the script-bound `otel` instance:
+ *
+ * def singleMBean = otel.mbean("com.example:type=SingleType")
+ * def multipleMBeans = otel.mbeans("com.example:type=MultipleType,*")
+ * [singleMBean, multipleMBeans].each { it.fetch() }
+ *
+ */
+class MBeanHelper {
+    private static final Logger logger = Logger.getLogger(MBeanHelper.class.getName());
+
+    private List<GroovyMBean> mbeans
+    protected JmxClient jmxClient
```

Comment:
Can these two `protected` fields be private? Direct field access is generally discouraged unless the class is just a barebones data class.

Reply:
Updated